kolibri/plugins/user_profile/tasks.py

```diff
@@ -29,6 +35,13 @@ def validate(self, data):
             job_data = super(MergeUserValidator, self).validate(data)
 
         job_data["kwargs"]["local_user_id"] = data["local_user_id"].id
+        if data.get("new_superuser_id"):
+            job_data["kwargs"]["new_superuser_id"] = data["new_superuser_id"].id
+
+        # create token to validate user in the new facility
+        # after it's deleted in the current facility:
+        token = TokenGenerator().make_token(data["username"])
```

Comment:
Can generate this after the sync has completed and add to the job metadata from inside the task. This will reduce the chance that the token expires after a slow sync, and means that we could use the id of the remote_user, and avoid needing to override to user the username.

Reply:
yep, thanks for the suggestion. Done too.